nova/scheduler/filters/shard_filter.py

```diff
@@ -148,18 +254,28 @@ def host_passes(self, host_state, spec_obj):
         if self._ALL_SHARDS in shards:
             LOG.debug('project enabled for all shards %(project_shards)s.',
                       {'project_shards': shards})
-            return True
         elif host_shard_names & set(shards):
             LOG.debug('%(host_state)s shard %(host_shard)s found in project '
                       'shards %(project_shards)s.',
                       {'host_state': host_state,
                        'host_shard': host_shard_names,
                        'project_shards': shards})
-            return True
         else:
             LOG.debug('%(host_state)s shard %(host_shard)s not found in '
                       'project shards %(project_shards)s.',
                       {'host_state': host_state,
                        'host_shard': host_shard_names,
                        'project_shards': shards})
             return False
+
+        if k8s_shard:
+            matches = any(host_shard == k8s_shard
+                          for host_shard in host_shard_names)
+            if not matches:
+                LOG.debug("%(host_state)s is not part of the requested "
+                          "K8S cluster shard '%(k8s_shard)s'",
```

Comment:
typo: since the user doesn't request a shard but a k8s cluster, imho this needs to be "requested K8S cluster's shard"

I'm also wondering if we shouldn't just ditch "requested" here and make it "is not part of the K8S cluster's shard"